packages/import-handler/src/matterHistory.ts

```diff
@@ -5,28 +5,234 @@
 
 import { parse } from '@fast-csv/parse'
 import { Stream } from 'stream'
+import unzip from 'unzip-stream'
+import fs from 'fs'
+import path from 'path'
+import * as fsExtra from 'fs-extra'
+
+import { parseHTML } from 'linkedom'
+import { Readability } from '@omnivore/readability'
+import createDOMPurify, { SanitizeElementHookEvent } from 'dompurify'
+
+import { encode } from 'urlsafe-base64'
+import crypto from 'crypto'
+import { ImportContext } from '.'
 
 export type UrlHandler = (url: URL) => Promise<void>
 
-export const importMatterHistory = async (
-  stream: Stream,
-  handler: UrlHandler
-): Promise<number> => {
+export const importMatterHistoryCsv = async (
+  ctx: ImportContext,
+  stream: Stream
+): Promise<void> => {
   const parser = parse({
     headers: true,
     strictColumnHandling: false,
   })
   stream.pipe(parser)
 
-  let count = 0
   for await (const row of parser) {
     try {
       const url = new URL(row['URL'])
-      await handler(url)
+      await ctx.urlHandler(ctx, url)
+      ctx.countImported += 1
     } catch (error) {
       console.log('invalid url', row, error)
+      ctx.countFailed += 1
+    }
+  }
+}
+
+const DOM_PURIFY_CONFIG = {
```

Comment:
I'm trying to find a way to move all these duplicate code in one place so we could reuse

Reply:
Yeah, would be nice, part of the issue is the other code is JS not typescript